### ISSUE TITLE
Can not login into Hassio with iOS 14 (18A5351d)

### ISSUE THREAD

**Issue by Smiggel** [home-assistant/frontend#6654]
## Checklist

- [x] I have updated to the latest available Home Assistant version.
- [x] I have cleared the cache of my browser.
- [x] I have tried a different browser to see if it is related to my browser.

## The problem
When trying to login into home assistant using a browser (Safari, Chrome and Firefox) on iOS, you can not login. The login screen keeps reloading even with correct credentials (work in a browser on a Mac and iOS 13).
After two tries you see the error: A problem has occurred repeatedly on: 'http://x.x.x.x:8123/auth/authorize?response_type=code&client_id=https://home-assistant.io/iOS&redirect_uri=homeassistant://auth-callback'

## Expected behavior
I would expect that the user name and password would be validated and I would see the dashboard being loaded.

## Steps to reproduce
1. Update HA to the latest version 0.114.2. 
2. Update iOS to version iOS 14 (18A5351d). 
3. Try to login with the browser of the mobile app.

## Environment
iPhone 8 + iOS 14 build 18A5351d
Safari native iOS browser
Firefox browser on iOS
Chrome browser on iOS

## Javascript errors shown in your browser console/inspector
No JS errors are visible.

http://x.x.x.x:8123/auth/authorize?response_type=code&redirect_uri=http%3A%2F%2Fx.x.x.x%3A8123%2F%3Fauth_callback%3D1&client_id=http%3A%2F%2Fx.x.x.x%3A8123%2F&state=eyJoYXNzVXJsIjoiaHR0cDovLzE5Mi4xNjguMC4xMDA6ODEyMyIsImNsaWVudElkIjoiaHR0cDovLzE5Mi4xNjguMC4xMDA6ODEyMy8ifQ%3D%3D

**Comment by TomBrien** [comment]
This appears to be a bug in iOS 14 beta 5. See home-assistant/iOS#925 for a little extra context.

**Comment by TomBrien** [comment]
Just for the benefit of anyone looking into this, Zac posted some extra details of the crash [on Discord](https://discordapp.com/channels/330944238910963714/351047592588869643/745746338582626375). Sadly this is also seen in the Technology Preview (aka beta) of the desktop Safari. 

```
0   com.apple.WebCore                 0x00000004d9abcde5 WebCore::EventHandler::dispatchMouseEvent(WTF::AtomString const&, WebCore::Node*, int, WebCore::PlatformMouseEvent const&, WebCore::EventHandler::FireMouseOverOut) + 2773
1   com.apple.WebCore                 0x00000004d8137d49 WebCore::EventHandler::handleMousePressEvent(WebCore::PlatformMouseEvent const&) + 3529
2   com.apple.WebKit                  0x000000010af0fc7d WebKit::WebPage::mouseEvent(WebKit::WebMouseEvent const&) + 475
…
```

**Comment by thanhtoan009** [comment]
Same error but can be logged. When entering username and password, click "return" button on ios keyboard.

**Comment by Smiggel** [comment]
> Same error but can be login. When entering username and password, click "return" button on ios keyboard.

Thanks! It helped me to login into the HA app again.

**Comment by tiems90** [comment]
> Same error but can be login. When entering username and password, click "return" button on ios keyboard.

Thanks! Worked for me as well with this workaround.

**Comment by alanranciato** [comment]
> Same error but can be login. When entering username and password, click "return" button on ios keyboard.

Awesome.  Been struggling trying to login.  Return worked!

**Comment by SConaway** [comment]
Happens on Safari in Big Sur too, but the workaround doesn't work there sadly (no biggie for me though).

**Comment by bramkragten** [comment]
Can someone check if it also happens in these demos: <https://mwc-demos.glitch.me/>

**Comment by SConaway** [comment]
@bramkragten what exactly would you like tested?

**Comment by TheJulianJES** [comment]
> https://mwc-demos.glitch.me/

They also crash, as soon as I hit a button.

**Comment by SeanPM5** [comment]
Yeah interacting with anything on that https://mwc-demos.glitch.me/ link crashes and reloads the web view for me on iOS 14 b5 and Safari Technology Preview on macOS Big Sur beta. 

So it's not a Home Assistant specific problem.

**Comment by SConaway** [comment]
Yep. Button crash.

**Comment by bramkragten** [comment]
Ok, so it is an upstream issue

**Comment by bramkragten** [comment]
If someone with the issue and logs could create an issue there, we can check if Apple or Google should fix

https://github.com/material-components/material-components-web-components

**Comment by Smiggel** [comment]
Submitted a bug report via the feedback of Apple. Hopefully it’s fixed soon.

**Comment by SlimGee** [comment]
Sure thanks

**Comment by SlimGee** [comment]
Thanks guys

**Comment by christianwgd** [comment]
I'm afraid you closed this too early. Have this bug with Safari 14.0 on macOS 10.15.7. Could anyone please state, where Apple fixed this? (Safari Version or so)

Thanks

**Comment by TheJulianJES** [comment]
@christianwgd It was fixed in one of the beta versions of Safari on BigSur (and on iOS).
However, Apple decided to push out a broken Safari 14 to Catalina.
I think someone is currently preparing a workaround.

**Comment by christianwgd** [comment]
Thanks to @TheJulianJES. I upgraded my home assistant to 0.115.3 which has solved my problem. Seems like the workaround is included there.
